doc/adrs/XXX_support-server-side-pagination-for-table.adoc

```diff
@@ -0,0 +1,108 @@
+= ADR-XXX Support Server-Side Pagination for Table Representation
+
+== Context
+
+The default pagination features of Material React Table are client-side.
+This means you have to have all of your data fetched and stored in the table all at once.
+This may not be ideal for large datasets.
+
+== Decision
+
+To address the limitations of client-side pagination, we will implement server-side pagination for Material React Table.
+This approach will allow us to fetch and display data more efficiently, especially for large datasets.
+The server-side pagination method will only be applied to table representations.
+For the use of tables as form widgets, client-side pagination will continue to be used.
+
+=== Material React Table
+
+To enable server-side pagination, we will set `manualPagination: true` in Material React Table.
+This will allow us to take control of pagination on the backend, ensuring that we can efficiently manage large datasets and accurately display the true total number of rows.
+The pagination must be managed within our own state, and listeners must be added to handle pagination state changes, allowing the backend to be called with the updated pagination data.
+
+[source, typescript]
+----
+const [pagination, setPagination] = useState({
+  pageIndex,
+  pageSize,
+});
+
+useEffect(() => {
+  //do something when the pagination state changes
+}, [pagination.pageIndex, pagination.pageSize]);
+
+const table = useMaterialReactTable({
+  onPaginationChange: setPagination,
+  state: { pagination },
+});
+
+
+----
+
+=== Frontend
+
+The pagination data will be added to the `representationId` of the subscription:
+
+[source, typescript]
+----
+`${tableId}?pageIndex=${pageIndex}&pageSize=${pageSize}`
+----
+
+=== GraphQL
+
+* Add pagination data in `Table` type
```

Comment:
I'm not sure we need to have this information on the table. What would be the need to this? The frontend already have this information.

Reply:
The frontend doesn't have the rowCount (total available result)

Reply:
Yes indeed, that's a good reason for this. The table representation needs to compute pagination data because it can never be sure to return the "whole table", it's just computing a subset visible by the frontend depending on the requested range.